source/connection-monitoring-and-pooling/tests/README.rst

```diff
@@ -0,0 +1,181 @@
+
+- ``version``: A version number indicating the expected format of the spec tests (current version = 1)
+- ``style``: A string indicating what style of tests this file contains. Currently ``unit`` is the only valid value
+- ``tests``: An array of tests that are to be run independently of each other
+
+Unit Test Format:
+=================
+
+All Unit Tests have some of the following fields:
+
+- ``description``: A text description of what the test is meant to assert
+- ``operations``: A list of operations to perform. Each operation has some of the following fields:
+
+  - ``name``: A string describing which operation to issue.
+  - ``args``: An array of arguments to pass to the operation
+  - ``thread``: The name of the thread in which to run this operation. If not specified, runs in the default thread
+  - ``object``: For some operation, specifies the object on which to run the operation
+  - ``returnTo``: For some operation, specifies what to name the return value of the operation
+
+- ``error``: Indicates that the main thread is expected to error during this test. Subfields (like ``message``) indicate what error is expected
+- ``events``: An array of all connection monitoring events expected to occur while running ``operations``
+- ``ignore``: An array of event names to ignore
+
+Valid Unit Test Operations are the following:
+
+- ``start(name)``: Starts a new thread named ``name``
+- ``wait(x)``: Sleep the current thread for x milliseconds
+- ``waitFor(name, suppressError?)``: wait for thread ``name`` to finish executing. If ``suppressError`` is true, ignore any errors coming from that thread, otherwise propagate the errors to the main thread.
+- ``returnTo = createPool(options)``: creates and returns a new Connection Pool with the specified options.
+
+  - ``enableConnectionMonitoring`` MUST be set to true, and any connection events MUST be captured.
+  - The returned pool must have an ``id`` properly set as if created by a server. For each test, the pool ``id`` should start at 0, and monotonically increase.
+  - The returned pool must have an ``address`` set as a string value.
+
+- ``returnTo = object.acquire()``: call ``acquire`` on Pool ``object``, returning the acquired connection
+- ``object.release(connection, force?)``: call ``release`` on Pool ``object``, passing in connection and optional force flag
+- ``object.clear()``: call ``clear`` on Pool ``object``
+- ``object.close()``: call ``close`` on Pool ``object``
+
+Spec Test Context References
+============================
+
+If an any point a value in a test is of the form ``{ $$ref: string[] }``, that value is considered a Context Reference. Context References are resolved by replacing the value with the current value of ``context[...$$ref]``
+
+The definition of RESOLVE in the Spec Test Runner is as follows:
+
+- RESOLVE takes two value, ``context`` and ``unresolved``
+- Notation is ``RESOLVE(context, unresolved)``
+
+Pseudocode implementation of ``RESOLVE(context, unresolved)``:
```

Comment:
What is a context exactly? Because reading the spec tests it seems that connection pool and connection objects are used as "contexts".